src/binary-writer.cc

```diff
@@ -1121,6 +1259,31 @@ void BinaryWriter::WriteExpr(const Func* func, const Expr* expr) {
       break;
     }
   }
+  return std::nullopt;
+}
+
+void BinaryWriter::WriteExpr(const Func* func, const Expr* top_expr) {
+  std::list<StateVar> stack;
```

Comment:
std::stack or std::deque

Reply:
also maybe we can cache it or make static, or even create some part of it on thread stack with manual ptr. Smth like SBO